cmd/shipperctl/cmd/count.go

```diff
@@ -0,0 +1,199 @@
+	if err != nil {
+		return err
+	}
+
+	namespaceList, err := kubeClient.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	var errList []string
+	countedReleases := []OutputRelease{}
+	for _, ns := range namespaceList.Items {
+		applicationList, err := shipperClient.ShipperV1alpha1().Applications(ns.Name).List(metav1.ListOptions{})
+		if err != nil {
+			errList = append(errList, err.Error())
+			continue
+		}
+		for _, app := range applicationList.Items {
+			contender, err := cmdutil.GetContender(&app, shipperClient)
+			if err != nil {
+				errList = append(errList, err.Error())
+				continue
+			}
+			trueClusters := cmdutil.FilterSelectedClusters(releaseutil.GetSelectedClusters(contender), clusters)
+			if len(trueClusters) == 0 {
+				counter++
+				countedReleases = append(
+					countedReleases,
+					OutputRelease{
+						Namespace: contender.Namespace,
+						Name:      contender.Name,
+					})
+			}
+		}
+	}
+
+	if printOption == "" {
+		cmd.Println("Number of *contenders* that are scheduled only on decommissioned clusters: ", counter)
+	} else {
+		printCountedRelease(cmd.OutOrStdout(), countedReleases)
+	}
+	if len(errList) > 0 {
+		return fmt.Errorf(strings.Join(errList, ","))
+	}
+	return nil
+}
+
+func runCountReleasesCommand(cmd *cobra.Command, args []string) error {
+	counter := 0
+
+	kubeClient, err := configurator.NewKubeClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	shipperClient, err := configurator.NewShipperClientFromKubeConfig(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+
+	namespaceList, err := kubeClient.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	var errList []string
+	countedReleases := []OutputRelease{}
+	for _, ns := range namespaceList.Items {
+		releaseList, err := shipperClient.ShipperV1alpha1().Releases(ns.Name).List(metav1.ListOptions{})
+		if err != nil {
+			errList = append(errList, err.Error())
+			continue
+		}
+		for _, rel := range releaseList.Items {
+			trueClusters := cmdutil.FilterSelectedClusters(releaseutil.GetSelectedClusters(&rel), clusters)
+			if len(trueClusters) == 0 {
+				counter++
+				countedReleases = append(
+					countedReleases,
+					OutputRelease{
+						Namespace: rel.Namespace,
+						Name:      rel.Name,
+					})
+			}
+		}
+	}
+
+	if printOption == "" {
+		cmd.Println("Number of *releases* that are scheduled only on decommissioned clusters: ", counter)
+	} else {
+		printCountedRelease(cmd.OutOrStdout(), countedReleases)
+	}
+	if len(errList) > 0 {
+		return fmt.Errorf(strings.Join(errList, ","))
+	}
+	return nil
+}
+
+func printCountedRelease(stdout io.Writer, outputReleases []OutputRelease) {
+	var err error
+	var data []byte
+
+	switch printOption {
+	case "yaml":
+		data, err = yaml.Marshal(outputReleases)
+	case "json":
+		data, err = json.MarshalIndent(outputReleases, "", "    ")
```

Comment:
Yaml prints things prettily, which is why I thought it can be useful. JSON is easily pipeable (for me) so that we can get owners' emails out of this list. 
In case the release is fully scheduled on decommissioned clusters, this command will not touch it if it is incumbent/contender. This means we need to contact those application owners and ask them to modify their application (re-rollout or delete application). JSON format helped me pipe application names to the service that gets owner emails.

Reply:
Fair enough, thanks for the context 👍